### PR TITLE
push gateway not found lookup result to the app level

### PR DESCRIPTION
**Pull request by jeffgrunewald**
pre-requisite to https://github.com/helium/blockchain-node/pull/181, allows the response from the blockchain-node to distinguish between a result with location and region information and when the ledger is unable to verify the gateway by the requested pubkey

### CODE REVIEW

File: src/service/follower.proto [hunk 1]
```diff
@@ -46,6 +38,23 @@ message follower_gateway_resp_v1 {
   region region = 7;
 }
 
+message gateway_not_found {}
```
Comment (andymck) [inline src/service/follower.proto:41]: might be better to include a generic error response which can cater to any future error handling ( including any additional APIs which get added over time)...something like:

enum follower_error {
  gateway_not_found = 0;
}

message follower_error_resp {
  follower_error error = 1;  
  bytes details = 2;
}

In the case of a gateway not found, the error would equal the gateway_not_found enum value, whilst details would be the pub key specified in the request

Comment (jeffgrunewald) [reply src/service/follower.proto]: done [54c1367](https://github.com/helium/proto/pull/209/commits/54c136709e9570979757fe2d1baf972bf98aa15a)

File: src/service/follower.proto [hunk 2]
```diff
@@ -7,6 +7,13 @@ import "blockchain_txn.proto";
 import "gateway_staking_mode.proto";
 import "region.proto";
 
+enum follower_error { gateway_not_found = 0; }
+
+message follower_error_resp {
```
Comment (madninja) [inline src/service/follower.proto:12]: Why is this here? What would bytes contain?

Comment (jeffgrunewald) [reply src/service/follower.proto:12]: See @andymck ’s previous comment; contextual but in this case would be used to send back the pubkey that was missing in case the caller was making multiple calls and needed to differentiate which gateway wasn’t found

Comment (madninja) [reply src/service/follower.proto:12]: Yeah, I really dislike the "blob of contextual data" approach here vs errors with actual details in the message type itself.

Comment (jeffgrunewald) [reply src/service/follower.proto:12]: I can remove it but I’m disinclined to gold-plate this since it’s such a short-lived solution. Lemme play around briefly with a more structured not_found error than just an enum

Comment (jeffgrunewald) [reply src/service/follower.proto]: adjusted here to make the error type more distinct: [0896e48](https://github.com/helium/proto/pull/209/commits/0896e488d62c86a2978b8546b268d5aa4608388d); lemme know what you think

**Review by madninja — APPROVED** [review]
